tests/User/AutoLoginMiddlewareTest.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+
+namespace Yiisoft\Yii\Web\Tests\User;
+
+use Nyholm\Psr7\Response;
+use PHPUnit\Framework\TestCase;
+use PHPUnit_Framework_MockObject_MockObject;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+use Yiisoft\Auth\IdentityInterface;
+use Yiisoft\Auth\IdentityRepositoryInterface;
+use Yiisoft\Yii\Web\User\User;
+use Yiisoft\Yii\Web\User\AutoLoginMiddleware;
+
+class AutoLoginMiddlewareTest extends TestCase
+{
+    /**
+     * @var RequestHandlerInterface|PHPUnit_Framework_MockObject_MockObject
+     */
+    private $requestHandlerMock;
+
+    /**
+     * @var ServerRequestInterface|PHPUnit_Framework_MockObject_MockObject
+     */
+    private $requestMock;
+
+    /**
+     * @var AutoLoginMiddleware
+     */
+    private $autoLoginMiddlewareMock;
+
+    /**
+     * @var IdentityRepositoryInterface
+     */
+    private $identityRepositoryInterfaceMock;
+
+    /**
+     * @var IdentityInterface
+     */
+    private $identityInterfaceMock;
+    /**
+     * @var User
+     */
+    private $userMock;
+
+    public function setUp(): void
+    {
+        parent::setUp();
```

Comment:
Please don't use class properties to mock some objects. 
Mock it directly in method that it needed

Reply:
That is, in each method mock what I need. It doesn't matter if I duplicate it in each method. What do I gain from this?

Reply:
You can do any useful acts in private methods and combain them like you want

Reply:
Updated in the last commit